src/python/WMCore/Services/DDM/DDM.py

```diff
@@ -0,0 +1,286 @@
+                    return False
+            except KeyError:
+                return False
+        return True
+
+
+class DDM(Service):
+    """
+    Class which provides client APIs to the DDM service.
+
+    Usage example:
+        api = 'copy'
+        ddmReq = DDMReqTemplate(api)
+        ddm = DDM()
+        ddm.makeRequest(ddmReq, api)
+    """
+
+    def __init__(self, url=None, logger=None, configDict=None):
+        """
+        configDict:
+            Dictionary with parameters that are passed to the super class.
+        """
+        url = url or "https://dynamo.mit.edu"
+        configDict = configDict or {}
+        configDict.setdefault('endpoint', url)
+        configDict.setdefault('cacheduration', 1)  # in hours
+        configDict.setdefault('accept_type', 'application/json')
+        configDict.setdefault('content_type', 'application/json')
+        configDict['logger'] = logger if logger else logging.getLogger()
+        super(DDM, self).__init__(configDict)
+        self['logger'].debug("Initializing DDM with url: %s", self['endpoint'])
+
+    def _getResult(self, ddmReq, apiUrl, callname=""):
+        """
+        Either fetch data from the cache file or query the data-service
+        :param metricNumber: a number corresponding to the SSB metric
+        :return: a dictionary
+
+        """
+
+        # TODO: to make this cache not per call but per DDM object instance
+        cachedApi = "%s.json" % callname
+
+        # striping the None values from the request before sending it to dynamo
+        ddmReq.strip()
+        msg = "INFO: sending data to DDM: "
+        msg += "API: {}: ddmReq: {}".format(ddmReq.api, pformat(ddmReq))
+        self['logger'].info(msg)
+
+        # The Request class expects this to be a dict, otherwise it tries to
+        # encode the inputdata once again and inserts parasite information and
+        # breaks the call to dynamo.
+        ddmReq = dict(ddmReq)
+        data = self.refreshCache(cachedApi,
+                                 apiUrl,
+                                 inputdata=ddmReq,
+                                 verb='POST')
+        results = data.read()
+        data.close()
+
+        results = json.loads(results)
+        return results
+
+    def makeRequest(self, ddmReq, api=None):
+        """
+        _makeRequest_
+
+        Makes a SINGLE request to DDM
+
+        :ddmReq: Request template an instance of DDMReqTemplate class.
+        :api:    The DDM api to be queried.
+        :return: Dictionary containing the result from DDM
+        """
+
+        # try to figure out the api from the ddmReq template
+        if not api:
+            api = ddmReq.api
+
+        if not api:
+            msg = "WARNING: skip sending data to DDM: "
+            msg += "API: {}: ddmReq: {}\n".format(ddmReq.api, pformat(ddmReq))
+            return None
+
+        apiUrl = "registry/request/{}".format(api)
+        callname = "DDM-{}".format(api)
```

Comment:
Is it for caching purposes? If so, then the `apiUrl` would suffice in my opinion.

Reply:
The default  value is because I am still for the idea to have a common cache for the whole module but not in a per request basis (of course it must be in the __init__ then). But if I do not make it work like that, then Yes, apiUrl makes more sense here. And of course having a default value her will be an error then - because we will overlap caches in case of a wrong call to the function.

Reply:
I tested it with apiUrl and it breaks with:
```
[Errno 2] No such file or directory: '/tmp/tivanov/.wmcore_cache_yq79dB/dynamo.mit.edu/5284144785438926808_POST_DDM-registry/request/pollcopy.json'
```
I am keeping it as it is for the moment.